fontir/src/ir.rs

```diff
@@ -189,7 +189,28 @@ impl GlyphPathBuilder {
     }
 
     pub fn qcurve_to(&mut self, p: impl Into<Point>) -> Result<(), PathConversionError> {
-        todo!("Support qcurve to {}", p.into());
+        // https://github.com/googlefonts/fontmake-rs/issues/110
+        // Guard clauses: degenerate cases
+        if self.is_empty() {
+            return self.move_to(p);
+        }
+        if self.offcurve.is_empty() {
+            return self.line_to(p);
+        }
+
+        // Insert an implied oncurve between every pair of offcurve points
+        for i in 0..(self.offcurve.len() - 1) {
+            // current offcurve to halfway to the next one
+            let lhs = self.offcurve[i];
+            let rhs = self.offcurve[i + 1];
+            let on = Point::new((lhs.x + rhs.x) / 2.0, (lhs.y + rhs.y) / 2.0);
+            self.path.quad_to(lhs, on);
+        }
+        // last but not least, the last offcurve to the provided point
+        self.path.quad_to(*self.offcurve.last().unwrap(), p.into());
+        self.offcurve.clear();
```

Comment:
I wonder if it might be helpful to write this whole thing as,


```rust
match &self.offcurve {
    [] => self.line_to(p),
    [one] => self.path.quad_to(one, p),
    multiple => { // your implied oncurve logic goes here
    },
}
self.offcurve.clear();
```

Reply:
I tried that - although I think I wrote it as match on .len, your way is nicer - and I felt it was less clear than having guards then the normal/expected case. Maybe that's just my pre-Rust habbits emerging though.

Reply:
Per IM this is a matter of taste more than this being the one true Rust way so I will leave it for now.